client/app/dbaas/logs/detail/aliases/link/aliases-link.controller.js

```diff
@@ -1,8 +1,137 @@
         this.serviceName = this.$stateParams.serviceName;
-        this.alias = this.$stateParams.aliasId;
+        this.LogsAliasesService = LogsAliasesService;
+        this.ControllerHelper = ControllerHelper;
+        this.LogsStreamsService = LogsStreamsService;
+        this.LogsIndexService = LogsIndexService;
+        this.CloudMessage = CloudMessage;
+        this.ServiceHelper = ServiceHelper;
+
+        this.initLoaders();
+    }
+
+    /**
+     * load data
+     *
+     * @memberof LogsAliasesLinkCtrl
+     */
+    initLoaders () {
+        this.availableStreams = this.$q.defer();
+        this.attachedStreams = this.$q.defer();
+        this.availableIndices = this.$q.defer();
+        this.attachedIndices = this.$q.defer();
+
+        this.alias = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.LogsAliasesService.getAliasWithStreamsAndIndices(this.serviceName, this.$stateParams.aliasId)
+                .then(alias => {
+                    if (alias.streams.length > 0) {
+                        this.selectedContent = this.contents[0].value;
+                    } else if (alias.indexes.length > 0) {
+                        this.selectedContent = this.contents[1].value;
+                    }
+                    this.attachedStreams.resolve(alias.streams);
+                    this.attachedIndices.resolve(alias.indexes);
+                    return alias;
+                })
+        });
+        this.alias.load();
+
+        this.streams = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.LogsStreamsService.getStreams(this.serviceName)
+        });
+        this.streams.load();
+
+        this.indices = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.LogsIndexService.getIndices(this.serviceName)
+        });
+        this.indices.load();
+
+
+        this.contents = this.LogsAliasesService.getContents();
+        if (this.$stateParams.defaultContent) {
+            this.selectedContent = this.$stateParams.defaultContent;
+        } else {
+            this.selectedContent = this.contents[0].value;
+        }
+
+        this.$q.all([this.alias.promise, this.streams.promise])
+            .then(result => {
+                const diff = _.filter(result[1], aapiStream => aapiStream.info.isEditable && !_.find(result[0].streams, attachedAapiStream => attachedAapiStream.info.streamId === aapiStream.info.streamId));
+                this.availableStreams.resolve(diff);
+            });
+
+        this.$q.all([this.alias.promise, this.indices.promise])
+            .then(result => {
+                const diff = _.filter(result[1], aapiIndex => aapiIndex.info.isEditable && !_.find(result[0].indexes, attachedAapiIndex => attachedAapiIndex.info.indexId === aapiIndex.info.indexId));
+                this.availableIndices.resolve(diff);
+            });
+    }
+
+    attachStream (items, defer) {
```

Comment:
This "defer" complicates things quite a bit. You should try to remove it. Here's how we would do it:

`this.LogsAliasesService.attachStream(this.serviceName, this.alias.data.info, items[0].info)
.catch(() => {
this.CloudMessage.error(this.$translate.instant("logs_aliases_attach_stream_fail", { stream: items[0].info.title }))
return this.$q.reject()};
`
Change to `return this.saveStream.load()` at the end of this fonction.

Also In the dual list component:
Around https://github.com/ovh-ux/ovh-manager-cloud/blob/feature/logs/client/app/ui-components/dual-list/dual-list.controller.js#L117 , change to `this.onAdd({ items: [item]}).catch(...)` (without using the defer)

Reply:
Same for remove.